inout.go

```diff
@@ -163,6 +163,26 @@ import "go.uber.org/dig"
 // Note that values in a value group are unordered. Fx makes no guarantees
 // about the order in which these values will be produced.
 //
+// To declare a soft relationship between a group an its constructors, use the
+// `soft` option on the group tag (`group:"[groupname],soft"`). A soft group
+// will be populated only with values from already-executed constructors, this
+// means constructors of soft value groups will be called only if they provide
+// another requested value.
```

Comment:
```suggestion
// will be populated only with values from already-executed constructors.
```

The last statement isn't quite true. If there's another function consuming the value then it will still populate it. 

It's better to describe soft value groups as specifying the parameter as an 'opportunistic' consumer - if the values are already provided in the DI container, it consumes them. Otherwise, it doesn't.

You also want to specify that this is only a valid option for input parameters in the docs.